CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift

```diff
@@ -69,20 +69,21 @@ class BreathlessViewController: UIViewController {
         
         tableView.rx.itemSelected
             .subscribe(onNext: {indexPath in
+                
                 if indexPath[1] == 0{
-                    print ("0")
+                    self.viewModel.onCauseSelected(cause: .leavingHouseOrDressing)
                 }
                 else if indexPath[1] == 1{
-                    print ("1")
+                    self.viewModel.onCauseSelected(cause: .walkingYardsOrMinsOnGround)
                 }
                 else if indexPath[1] == 2{
-                    print ("2")
+                    self.viewModel.onCauseSelected(cause: .groundOwnPace)
                 }
                 else if indexPath[1] == 3{
-                    print ("3")
+                    self.viewModel.onCauseSelected(cause: .hurryOrHill)
                 }
                 else if indexPath[1] == 4{
-                    print ("4")
+                    self.viewModel.onCauseSelected(cause: .exercise)
```

Comment:
The array with the items should be in the view model. In another variable, in the view model, expose it to the view controller (using `Driver`, like in other view models). Here pass only the index: `viewModel.onCauseSelected(indexPath[1])`. In the view model, access the array using the index to get cause.

This is an acceptable solution. I personally prefer to not pass indices around but that would be more involved.

Also, you can declare `viewModel` in the capture list of the closure, to not have to use `self`: `.subscribe(onNext: { [viewModel] indexPath in`

Reply:
General takeaways:
- Don't put data and logic in the UI (which includes view controllers). Thus view "model" - it's the backing model for the UI.
- If you're repeating blocks of code for each index/cell, like here, you probably want to access an array instead.
- Orient with existing code: The communication pattern between view models and view controllers is/should be the same everywhere.

Reply:
sounds good! will do!